spec/duration.html

```diff
@@ -1184,7 +1176,7 @@ <h1>
       <emu-alg>
         1. If _relativeTo_ is not present, set _relativeTo_ to *undefined*.
         1. If Type(_relativeTo_) is Object and _relativeTo_ has an [[InitializedTemporalZonedDateTime]] internal slot, then
-          1. Let _endNs_ be ? AddZonedDateTime(_relativeTo_.[[Nanoseconds]], _relativeTo_.[[TimeZone]], _relativeTo_.[[Calendar]], 0, 0, 0, _days_, _hours_, _minutes_, _seconds_, _milliseconds_, _microseconds_, _nanoseconds_).
+          1. Let _endNs_ be ? AddZonedDateTime(_relativeTo_.[[Nanoseconds]], _relativeTo_.[[TimeZone]], _relativeTo_.[[Calendar]], ! CreateTemporalDurationRecord(0, 0, 0, _days_, _hours_, _minutes_, _seconds_, _milliseconds_, _microseconds_, _nanoseconds_)).
```

Comment:
Nit: Typo `CreateTemporalDurationRecord` should be `CreateDurationRecord` throughout this commit.

`CreateDurationRecord` is fallible, because `days` can have a different sign than the time components:

```js
let d = Temporal.Duration.from({
  weeks: 1,
  days: 0,
  hours: 1,
});

let cal = new class extends Temporal.Calendar {
  #dateAdd = 0;
  dateAdd(date, duration, options) {
    if (++this.#dateAdd === 1) {
      duration = "-P1W";
    }
    return super.dateAdd(date, duration, options);
  }
}("iso8601");

let zdt = new Temporal.ZonedDateTime(0n, "UTC", cal);
let r = d.total({
  relativeTo: zdt,
  unit: "days",
});
console.log("rounded result=", r);
```

Reply:
I updated this spot in BalanceDuration where the changes were introducing a risk, but am rather confident that the other invocations are safe. I'm less confident that the use of CreateTemporalDuration within AddZonedDateTime is safe, but that is not changed by this PR.